lib/ttfunk/table/dsig.rb

```diff
@@ -0,0 +1,48 @@
+module TTFunk
+  class Table
+    class Dsig < Table
+      class SignatureRecord
+        attr_reader :format, :length, :offset, :signature
+
+        def initialize(format, length, offset, signature)
+          @format = format
+          @length = length
+          @offset = offset
+          @signature = signature
+        end
+      end
+
+      attr_reader :version, :flags, :signatures
+
+      TAG = 'DSIG'.freeze
+
+      def self.encode(dsig)
+        return nil unless dsig
+
+        # Don't attempt to re-sign or anything - just use dummy values.
+        # Since we're subsetting that should be permissible.
+        [dsig.version, 0, 0].pack('Nnn')
+      end
+
+      def tag
+        TAG
+      end
+
+      private
+
+      def parse!
+        @version, num_signatures, @flags = read(8, 'Nnn')
```

Comment:
I wonder if we want to spend time parsing the table if we would never use the results? As far as I can tell we only ever use version and nothing else.

Reply:
That's true. We do this in a number of other tables (parse values we never use). I assumed that was because the tables are all accessible via the `File`s public interface, so any client of the gem might want or need to retrieve font information the library itself doesn't use.